### DISCUSSION TITLE
JW-1: Bacterial Classifiers

### DISCUSSION THREAD

**Discussion by jweiner12** [Global-Chem/global-chem#98] [Ideas]
Hey guys, I'm looking for some ways we can apply the bacterial classifiers. I think the easiest will be to use them as a database for pathogenic and/or non-pathogenic bacteria where you would enter in the genus or species name and get back all the relevant classifiers for that entry. I think we should start off by just focusing on bacterial genuses instead of the individual species for a more general classification and then later we can get more specific. To start off we will need to compile a list of bacterial genuses especially those that are pathogenic and currently have a lot of research being done on them. I can then work on classifying them based on the various classifiers that we have. Once we have the most relevant bacteria in there we can work on adding less relevant and non-pathogenic bacteria. If anyone works with or knows people that work with a specific bacterial type throw it in here and I'll be sure to add it to the list.

**Comment by Sulstice** [comment]
Hey Jake, 

I realized that as well when mining your notes you highlighted a lot of relevant bacteria that can be used for different models. A good way to compile the list would be to do into a excel file or csv/tsv file and keep adding to the list on github. Github has a "Pretty View" which will make it easier to edit, I got you started on one here:

https://github.com/Sulstice/global-chem/blob/master/global_chem_extensions/global_chem_extensions/bioinformatics/applications/bacteria.csv

<img width="947" alt="Screen Shot 2022-05-03 at 6 32 49 AM" src="https://user-images.githubusercontent.com/11812946/166438956-d235b52d-c564-4aba-bf38-df199dedbefd.png">

Any classifiers that are not used we can say "n/a", this can help us keep track of stuff.  To initialize bacteria models, code wise, they can be classified in the following ways based on your notes:

```python

bacteria = Bacteria(
        name='staphylococcus',
        cell_wall_composition='thick_peptidoglycan',
        genus_name = '',
        species_name = '',
        family = 'chlorobiaceae',
        gram_positive = False,
        gram_negative = False,
        inner_membrane = None,
        outer_membrane = None,
        shape = None,
        nutrition_type = None,
        respiratory = None,
        environment = None,
        antisera = None,
        universal_phylogenetic_tree_rna_sequence = None,
        taxonomic_hierarchy = None,
 )
```

Data storage and feasibility wise we can use the sheet to build the models appropriately for people to use instead of always writing their own.
